lib/event_tracer/datadog_logger.rb

```diff
@@ -64,9 +67,20 @@ def valid_args?(metrics)
     end
 
     def build_tags(args)
-      args.slice(*allowed_tags).map do |tag, value|
+      args.slice(*(allowed_tags + default_tags)).map do |tag, value|
         "#{tag}:#{value}"
       end
     end
+
+    def environment
+      ENV['APP_ENV']
+    end
+
+    def append_default_tags(args)
+      # Make sure each default tag is appended
+      @default_tags.each do |tag|
+        args[tag] = send(tag)
+      end
+    end
```

Comment:
Why don't we use the code in the ticket?
```ruby
def build_tags(args)
  default_tags.merge(args.slice(*allow_tags)).map do |tag, value|
    "#{tag}:#{value}"
  end
end
```

This allows each call to overwrite the default tag if necessary. Imagine you have this:

```ruby
# we already set `ENV['APP_ENV'] = 'uat'`
EventTracer.info("message", action: Action, metrics: [:metric], tags: { environment: 'staging' })
```
Just reading the code without knowing about the implementation detail of the gem, what would you expect? Generally, code of a library should follow the rule of least surprise.

Reply:
Okay, i didn't know that you imagined default_tags as a hash. In my/our original conception it was just a list of tags without specific values 👍